src/core/mod.rs

```diff
@@ -21,34 +21,156 @@ use parsec_interface::operations::psa_key_attributes::KeyAttributes;
 use parsec_interface::operations::psa_sign_hash::Operation as PsaSignHash;
 use parsec_interface::operations::psa_verify_hash::Operation as PsaVerifyHash;
 use parsec_interface::operations::{NativeOperation, NativeResult};
-use parsec_interface::requests::Opcode;
-use parsec_interface::requests::ProviderID;
 use std::collections::HashSet;
 
+pub use parsec_interface::operations::{psa_algorithm, psa_key_attributes};
+pub use parsec_interface::requests::{BodyType, Opcode, ProviderID};
+
 /// Core client for Parsec service
 ///
-/// The client exposes low-level functionality for using the Parsec service
+/// The client exposes low-level functionality for using the Parsec service.
+/// Below you can see code examples for a few of the operations supported.
+///
+/// Creating a `CoreClient` instance:
+///```no_run
```

Comment:
No :( because it tries to connect to the default socket... if I want to make it run I have to copy over all the code around the mock core client and the mock socket....

Reply:
Ah yes forgot about the connection part, it's fine then!